src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/TopologyHelper.groovy

```diff
@@ -133,6 +133,15 @@ class TopologyHelper {
         return topology.getActiveSwitches().shuffled().first()
     }
 
+    SwitchTriplet findSwitchTripletWithAlternativesPaths() {
+        return switchTriplets.find {
+            if (it.ep1 == it.ep2 || it.ep1 == it.shared || it.ep2 == it.shared) {
```

Comment:
It would be great if you move this condition to closure field e.g. in SwitchTriplet, or into new SwitchTripletFilters class. So it'll look like this:
```
class SwitchTriplet {
static Closure ALL_ENDPOINTS_DIFFERENT = {SwitchTriplet swT -> swT.ep1 != swT.ep2 && swT.ep1 != swT.shared && swT.ep2 == swT.shared}
...
```
And in this class or even in test:
return switchTriplets.findAll(ALL_ENDPOINTS_DIFFERENT).findAll{...}

Reply:
Agree about the performance. But these are [usually] manipulations in memory, so we may don't care about noticeable performance degradation.
Idea is to have small 'bricks' which can be reused/combined into complicated (and, what is important, readable) conditions, like `.findAll(ALL_ENDPOINTS_DIFFERENT).findAll(HAS_THREE_DIVERSE_PATHS).find{ONE_SWITCH_IS_CENTEC}